### ISSUE TITLE
form->getMessages() dose not contain all childs (fieldset) messges

### ISSUE THREAD

**Issue by Baft** [zendframework/zend-form#15]
`filedset->has()` , dose not check all childs (fieldsets) recursivley .

form access elements is top-down like : `$form->get('fieldset1')->get('fieldset2')->get('element1')`  and can not  access it directly like : `$form->get('element1')` )  .
it is expected by parents to  access childs elements( for ease of use).
also `form->setMessages()` can not save all child fieldsets messages. because of that .
so `form->getMessaged()` dose not contain all messages 
## test

with form config like this :

```
[
'hydrator' => 'Zend\Stdlib\Hydrator\ArraySerializable',
    'name' => 'from1',
    'fieldsets' => [
        'fldset1' => [
            'flags' => [
                'name' => 'fldset1'
            ],
            'spec' => [
                'fieldsets' => [
                    'fldset2'=> [
                        'flags' => [
                            'name' => 'fldset2'
                        ],
                        'spec' => [
                            'elements' => [
                                [
                                    'flags' => [
                                        'name' => 'fldset2_el1'
                                    ],
                                    'spec' => [
                                        'type' => 'text',
                                        'name' => 'fldset2_el1'
                                    ]
                                ]
                            ]//elements close
                        ]
                    ]//fldset2 close
                ],
                'elements' => [
                        [
                            'flags' => [
                                'name' => 'fldset1_el1'
                            ],
                            'spec' => [
                                'type' => 'text',
                                'name' => 'fldset1_el1'
                            ]
                        ]
                ]//elements close
            ]
        ]//fldset1 close
    ],
    'elements' => [
            [
                    'flags' => [
                            'name' => 'el1'
                    ],
                    'spec' => [
                            'type' => 'text',
                            'name' => 'el1'
                    ]
            ]
    ],//elements close
    'input_filter' => array (
            'el1' => array (
                    'name' => 'el1',
                    'required' => true,
                    'allow_empty' => false,
                    'validators' => array (
                            array (
                                    'name' => 'Zend\Validator\StringLength',
                                    'options' => array (
                                            'min' => 6,
                                            'max' => 256,
                                    )
                            )
                    )
            ),
            'fldset1_el1' => array (
                    'name' => 'fldset1_el1',
                    'required' => true,
                    'allow_empty' => false,
                    'validators' => array (
                            array (
                                    'name' => 'Zend\Validator\StringLength',
                                    'options' => array (
                                            'min' => 6,
                                            'max' => 256,
                                    )
                            )
                    )
            ),
            'fldset2_el1' => array (
                    'name' => 'fldset2_el1',
                    'required' => true,
                    'allow_empty' => false,
                    'validators' => array (
                            array (
                                    'name' => 'Zend\Validator\StringLength',
                                    'options' => array (
                                            'min' => 6,
                                            'max' => 256,
                                    )
                            )
                    )
            ),
    )
]//close
```

check :

```
$tstForm=(new Factory())->createForm($formConfig);
$tstForm->setData([
                "el1"=>'',
                "fldset1_el1"=>'',
                "fldset2_el1"=>''
        ])->isValid();

// just return el1 error message , not other element . expexted to be contain all messges of form
var_dump($tstForm->getMessages());

//true , direct child (one level) 
var_dump($tstForm->has('el1'));

//flase , true . 'has' function dose not check form recursivley
var_dump($tstForm->has('fldset1_el1') , $tstForm->get('fldset1')->has('fldset1_el1'));
```

**Comment by adamlundrigan** [comment]
This is the intended behaviour.  `Fieldset::has` is intended to only check for existence of a direct descendant of the fieldset.

In your check script you would set your data array structure to match the hierarchy of fieldsets, eg:
```php
$tstForm->setData([
    'fldset1' => [
        'fldset2' => [
            'fldset2_el1' => 'value',
        ],
        'fldset1_el1' => 'value',
    ],
    'el1' => 'value',
]);
```

**Comment by Baft** [comment]
this is logistic to check only direct descendant .
it beseem in viewpoint of api design  to be able use simply and match with most usecase (most usecases, pass form value without knowing about hierarchy of fieldsets ) in addition to the standards and advanced usecases (like your script on set data and when pass from model  ).
it beseem good if we have ability on fieldset and form object to check elements value and finding elements in both case (recursivley or direct).

**Comment by froschdesign** [comment]
This should fixed with #136
